lib/algolia/index.rb

```diff
@@ -264,6 +264,56 @@ def get_objects(objectIDs, attributes_to_retrieve = nil, request_options = {})
       client.post(Protocol.objects_uri, { :requests => requests }.to_json, :read, request_options)['results']
     end
 
+    #
+    # find_object search iteratively through the search response `hits`
+    # field to find the first response hit that would match against the given
+    # `filter_func` function.
+    #
+    # If no object has been found within the first result set, the function
+    # will perform a new search operation on the next page of results, if any,
+    # until a matching object is found or the end of results, whichever
+    # happens first.
+    #
+    # To prevent the iteration through pages of results, `paginate`
+    # parameter can be set to false. This will stop the function at the end of
+    # the first page of search results even if no object does match.
+    #
+    def find_object(filter_func, request_options = {})
```

Comment:
Is `filter_func` idiomatic in ruby?

Reply:
We may have to adapt this method, as the `search` as search_parameters and request_options. Any idea?